README.md

```diff
@@ -156,10 +156,10 @@ Prerequities:
 module "egress_proxy" {
   source = "github.com/GSA-TTS/terraform-cloudgov//egress_proxy?ref=v2.0.0-beta.1"
 
-  cf_org_name     = local.cf_org_name
-  cf_egress_space = data.cloudfoundry_space.egress_space
-  cf_client_space = data.cloudfoundry_space.app_space
-  name            = "egress-proxy"
+  cf_org_name      = local.cf_org_name
+  cf_egress_space  = data.cloudfoundry_space.egress_space
+  cf_client_spaces = {(data.cloudfoundry_space.app_space.name) = data.cloudfoundy_space.app_space.id}
+  name             = "egress-proxy"
   allowlist = {
     "source_app_name" = ["host.com:443", "otherhost.com:443"]
```

Comment:
Since we're now allowing multiple client spaces, the app names specified as keys may not be unique. We should probably set a new naming convention that will keep those keys useful for distinguishing between clients later while still being easy to read, eg:
```suggestion
    "source_space_name/source_app_name" = ["host.com:443", "otherhost.com:443"]
```

In this case, maybe we should get the list of space names from those indices and just drop the `cf_client_space(s)` variable altogether as redundant.

Reply:
Agreed on revisiting this. I am avoiding just passing space names around because you end up with nasty race conditions where terraform thinks it can operate on a space but that space does not yet exist and you have to add a ton of `depends_on = []` to your root module

Reply:
It looks like it's only used for [creating an UPSI with client credentials in each space where clients exist](https://github.com/GSA-TTS/terraform-cloudgov/blob/a11cbd14c8fe694778dcf734c135fa1798728298/egress_proxy/main.tf#L76-L89). I think a better solution is for the module to do less: just return the credential JSON in an output variable, and make it the caller's job to decide the best way for that information to be used. (They may well choose to create an UPSI with that content in each client space, or they may grab values out and swizzle them into some other form, pass info via env vars, etc.)

Reply:
(That has the side benefit of resolving dependency cycles, because if the UPSIs are created outside the module, Terraform has much better control of implicit dependencies with the spaces.)